contracts/OgvStaking.sol

```diff
@@ -108,54 +125,77 @@ contract OgvStaking is ERC20Votes {
     /// @param amount OGV to lockup in the stake
     /// @param duration in seconds for the stake
     function stake(uint256 amount, uint256 duration) external {
-        _stake(amount, duration, msg.sender);
+        revert StakingDisabled();
     }
 
-    /// @dev Internal method used for public staking
-    /// @param amount OGV to lockup in the stake
-    /// @param duration in seconds for the stake
-    /// @param to address to receive ownership of the stake
-    function _stake(uint256 amount, uint256 duration, address to) internal {
-        require(to != address(0), "Staking: To the zero address");
-        require(amount <= type(uint128).max, "Staking: Too much");
-        require(amount > 0, "Staking: Not enough");
-
-        // duration checked inside previewPoints
-        (uint256 points, uint256 end) = previewPoints(amount, duration);
-        require(points + totalSupply() <= type(uint192).max, "Staking: Max points exceeded");
-        _collectRewards(to);
-        lockups[to].push(
-            Lockup({
-                amount: uint128(amount), // max checked in require above
-                end: uint128(end),
-                points: points
-            })
-        );
-        _mint(to, points);
-        ogv.transferFrom(msg.sender, address(this), amount); // Important that it's sender
-
-        if (!hasDelegationSet[to] && delegates(to) == address(0)) {
-            // Delegate voting power to the receiver, if unregistered
-            _delegate(to, to);
-        }
+    /// @notice Collect staked OGV for a lockup and any earned rewards.
+    /// @param lockupId the id of the lockup to unstake
+    /// @return unstakedAmount OGV amount unstaked
+    /// @return rewardCollected OGV reward amount collected
+    function unstake(uint256 lockupId) external returns (uint256 unstakedAmount, uint256 rewardCollected) {
```

Comment:
Big thumbs up on driving all the external methods through a single internal method. Much cleaner code.